oauthproxy.go

```diff
@@ -114,14 +114,24 @@ func NewOAuthProxy(opts *options.Options, validator func(string) bool) (*OAuthPr
 		}
 	}
 
+	providerNameArray := make([]string, 0)
+	providerIDArray := make([]string, 0)
+
+	for i := range opts.Providers {
+
+		providerNameArray = append(providerNameArray, opts.Providers[i].Name)
+		providerIDArray = append(providerIDArray, opts.Providers[i].ID)
+	}
+
 	pageWriter, err := pagewriter.NewWriter(pagewriter.Opts{
 		TemplatesPath:    opts.Templates.Path,
 		CustomLogo:       opts.Templates.CustomLogo,
 		ProxyPrefix:      opts.ProxyPrefix,
 		Footer:           opts.Templates.Footer,
 		Version:          VERSION,
 		Debug:            opts.Templates.Debug,
-		ProviderName:     buildProviderName(opts.GetProvider(), opts.Providers[0].Name),
```

Comment:
Does it make sense to rework this again so that every provider can have a custom override field?